lib/screens/chat_screens/chat_sceen.dart

```diff
@@ -0,0 +1,255 @@
+              IconButton(icon: Icon(Icons.close), onPressed: () {}),
+            ],
+            backgroundColor: Colors.lightBlueAccent,
+          ),
+          title: recipient),
+      body: SafeArea(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: <Widget>[
+            MessagesStream(groupChatId: groupChatId, fromId: fromId),
+            Container(
+              width: double.infinity,
+              color: Colors.white,
+              padding:
+                  EdgeInsets.only(left: 10, bottom: 10, top: 10, right: 10),
+              child: Row(
+                crossAxisAlignment: CrossAxisAlignment.center,
+                children: <Widget>[
+                  Expanded(
+                    child: TextField(
+                      keyboardType: TextInputType.multiline,
+                      controller: messageTextController,
+                      onChanged: (value) {
+                        messageText = value;
+                      },
+                      textCapitalization: TextCapitalization.sentences,
+                      decoration: InputDecoration(
+                          hintText: "Write message...",
+                          hintStyle: TextStyle(color: Colors.black54),
+                          border: InputBorder.none),
+                      maxLines: null,
+                    ),
+                  ),
+                  SizedBox(
+                    width: 15,
+                  ),
+                  FlatButton(
+                    shape: RoundedRectangleBorder(
+                        borderRadius: BorderRadius.circular(18.0)),
+                    color: Colors.cyan,
+                    onPressed: () {
+                      messageTextController.clear();
+                      var currentTimeAndDate = DateTime.now();
+                      //message text + loggedInuser.email
+
+                      _firestore
+                          .collection('messages')
+                          .doc(groupChatId)
+                          .collection(groupChatId)
+                          .add({
+                        'text': messageText.trim(),
+                        'sender': loggedInUser.email,
+                        'timestamp': currentTimeAndDate,
+                        'idFrom': userId,
+                        'idTo': fromId,
+                      });
+                    },
+                    child: Text(
+                      'Send',
+                      style: TextStyle(color: Colors.white),
+                    ),
+                  ),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class MessagesStream extends StatelessWidget {
+  final String groupChatId;
+  final String fromId;
+  MessagesStream({this.groupChatId, this.fromId});
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
I don't think you'll need a full class for this, you can just a custom widget if I'm understanding this workflow.